dataland-frontend/tests/component/components/resources/dataTable/conversion/NumberValueGetterFactory.cy.ts

```diff
@@ -1,41 +1,63 @@
 import {
-  EmptyDisplayValue,
-  MLDTDisplayComponents,
-  type MLDTDisplayValue,
-} from "@/components/resources/dataTable/MultiLayerDataTableCells";
+  MLDTDisplayObjectForEmptyString,
+  MLDTDisplayComponentName,
+  type MLDTDisplayObject,
+} from "@/components/resources/dataTable/MultiLayerDataTableCellDisplayer";
 import { numberValueGetterFactory } from "@/components/resources/dataTable/conversion/NumberValueGetterFactory";
+import { type Field } from "@/utils/GenericFrameworkTypes";
 
 describe("Unit test for the NumberValueGetterFactory", () => {
+  const sampleNumberFormFieldWithUnit: Field = {
+    name: "numberFormFieldTest",
+    label: "Some Label For Number Field",
+    component: "NumberFormField",
+    showIf: () => true,
+    description: "This is a dummy field for the test",
+    unit: "someUnit",
+  };
+
+  const sampleNumberFormFieldWithoutUnit: Field = { ...sampleNumberFormFieldWithUnit };
+  sampleNumberFormFieldWithoutUnit.unit = undefined;
+
   it("An empty string should be displayed if the data point is undefined", () => {
     const dataset = { data: undefined };
-    const value = numberValueGetterFactory("data")(dataset);
-    expect(value).to.deep.equal(EmptyDisplayValue);
+    const value = numberValueGetterFactory("data", sampleNumberFormFieldWithUnit)(dataset);
+    expect(value).to.deep.equal(MLDTDisplayObjectForEmptyString);
   });
 
-  it("The value of the input should be displayed if it exists", () => {
+  it("The value of the input should be displayed together with its unit if it exists", () => {
```

Comment:
Do we need this test? We also check for the unit in the test below.

Reply:
I haven't added this test, it already exists in main. But if you want my opinion on it:

Yes the unit is also checked in the test below. But the test below is even a little "stricter" because it additionally relies on the rounding to work as expected.

So if the unit is displayed correctly, but the rounding is broken, this test here will pass, but the one below will fail, telling you that the rounding must be broken.

Reply:
ok